### DISCUSSION TITLE
For local development, can I make auth process disable?

### DISCUSSION THREAD

**Discussion by SPONGE-JL** [microcks/microcks#873] [Q&A] [answered]
Hi team,

For local development, I want to set standalone microcks. So we need to hava [a set of components: mongo, keycloak, postman, app](https://github.com/microcks/microcks/blob/master/install/docker-compose/docker-compose.yml).

But in my local development environment, the authentication process is useless and the resources servedto keycloak seems that is not nessasary. Just I want to access directly the dashboards. 🥹

So, how do I make it disable? In my think, the authentication is need only when running in DEV, STAGE, PROD, etc.

(p.s) If the answer is "CANNOT DOING THAT", why we need it?

Thank you for reading.

Regards,

**Comment by lbroudoux** [comment, marked as answer]
Hey there, 

Keycloak has been made optional other time and can be disabled when it makes sense (and local development is one of those case 😉 ). Actually, it is already disabled when using Microcks through our [Docker Desktop Extension](https://microcks.io/blog/docker-desktop-extension-0.2/)

When using docker compose, you can remove the `keycloak` service and just have to add an extra `environment` on `app` container that states `KEYCLOAK_ENABLED=false`. This is actually what is done in `docker-compose-devmode.yml` (but it brings out Red Panda + Async features). 

We're even working on a lighter distribution with no dependency on MongoDB for local laptop development purposes (see microcks/microcks#851).

Hope it helps,

**Reply by SPONGE-JL** [reply]
Thanks @lbroudoux !
That is simple way and your activity (issue #851) is great! I also follow it :)

Have a nice weekend!

**Reply by SPONGE-JL** [reply]
In fact, I already used the [Docker Desktop Extension](https://microcks.io/blog/docker-desktop-extension-0.2/). But my needs is using multiple repositories control. Here is example as what I want.

Repositories in local environment
1. microservice-a : use a standalone lightweight microcks container on port `9103`
2. microservice-b : use another standalone lightweight microcks container on port `9203`

So your answer is simple way to implement my want.
Thanks a lot again, @lbroudoux :)

**Reply by lbroudoux** [reply]
Just for my curiosity: couldn't you load both microservices in same Microcks instance? Do you absolutely need different ports?

**Reply by SPONGE-JL** [reply]
Good suggestion. Yes, it could be!
In currently I focus on separation context about each microservice.

For example, a Staff Engineer support some troubles(A) about microservice-a. In that time, his local environment has already supported another troubles about microservice-b(B). Even if their A & B contexts does not overlap each other, but I feel easy to handle sepatate containers. And after the done of 'B', it is easy to remove all contexts ans states without concern and relation about 'A'.

I expect the situation about expanding to multiple teams and the temporary collaboration with outside of each teams.

**Reply by lbroudoux** [reply]
Got it. Thanks for the explanation. 

FYI, we have users in the community that are doing this (adding / removing services definition on the fly) using the Microcks API with isolated definition files. We also aim to enhance the CLI for that use-case so it will allow easier isolated fine grained addition/removal with fewer resource consumption.

**Reply by SPONGE-JL** [reply]
Wow cool, that approach would give team more smater and faster collaboration experience! 😆
I'm glad I asked the this questions. 🌟

**Reply by SPONGE-JL** [reply]
I use the CLI on this case, and head with [an issue on `microcks-cli`](https://github.com/microcks/microcks-cli/issues/27).

@lbroudoux , the CLI what you called is that, isn't it?

**Reply by lbroudoux** [reply]
Yes exactly but as of today the CLI doesn't support un-authenticated calls. This is something that is on our TODO, unfortunately. As of today, you need to go through an API call using `curl` command or whatever. Sorry.

**Reply by SPONGE-JL** [reply]
Nope, no to say Sorry, your insight helpful :)
I will try using `curl` you told. Thanks alot @lbroudoux 😍
